src/metabase/driver/mysql.clj

```diff
@@ -313,6 +314,12 @@
    ;; GZIP compress packets sent between Metabase server and MySQL/MariaDB database
    :useCompression       true})
 
+(defn- add-program-name-option [jdbc-spec]
+  ;; connectionAttributes (if multiple) are separated by commas, so values that contain spaces are OK, so long as they
+  ;; don't contain a comma; our mb-version-and-process-identifier shouldn't contain one, but just to be on the safe side
+  (let [prog-name (str/replace config/mb-version-and-process-identifier "," "_")]
+    (assoc jdbc-spec :connectionAttributes (str "program_name:" prog-name))))
```

Comment:
If someone specifies `connectionAttributes` in the additional JDBC options, this would override that, wouldn't it? Are there any other `connectionAttributes` one might specify? I wouldn't want to break things if someone had been setting this in the past and then we started stomping on that

Reply:
You might want to look for that and not set it if it's in the additional options

Reply:
This is now handled (new test added)